addons/project/tests/test_task_state.py

```diff
@@ -155,3 +155,25 @@ def test_duplicate_task_state_retention_with_open_dependencies(self):
 
         self.assertEqual(self.task_1.state, '04_waiting_normal')
         self.assertEqual(task_1_copy.state, '04_waiting_normal')
+
+    def test_state_dont_reset_when_enabling_task_dependencies(self):
+        self.task_1.state = "03_approved"
+        self.task_2.state = "02_changes_requested"
+        self.env['res.config.settings'].create({'group_project_task_dependencies': True}).execute()
+        self.assertEqual(self.task_1.state, "03_approved")
+        self.assertEqual(self.task_2.state, "02_changes_requested")
+
+    def test_recompute_state_when_task_dependencies_feature_changes(self):
+        self.env['res.config.settings'].create({'group_project_task_dependencies': True}).execute()
```

Comment:
maybe a few comment might help understand the flow of the test. The evolution of states can be obscure if you're not used to the feature.

Reply:
Indeed, I will add that. 🙂